elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ToElasticsearch.java

```diff
@@ -255,6 +259,20 @@ private static JsonObject convertMatchAllDocsQuery(MatchAllDocsQuery matchAllDoc
 		return JsonBuilder.object().add( "match_all", new JsonObject() ).build();
 	}
 
+	private static JsonObject convertMatchNoDocsQuery(MatchNoDocsQuery matchNoDocsQuery) {
```

Comment:
Probably not worth it for this but at one point I suppose we will need to inject the dialect in the conversion mechanism.